src/pybind/mgr/volumes/fs/operations/versions/subvolume_base.py

```diff
@@ -712,43 +653,39 @@ def snapshot_visibility_set(self, value):
         subvol_root_path = os.path.dirname(self.path)
         subvol_v2_path = self.path
         snaps_visibility_vxattr = "ceph.dir.subvolume.snaps.visible"
-        subvolume_size = 0
         try:
             self.fs.setxattr(subvol_root_path, snaps_visibility_vxattr,
                              str(value).encode('utf-8'), 0)
         except cephfs.Error as e:
             raise VolumeException(-e.args[0], e.args[1])
 
-        # in case of a sized subvolume, a new srnode will be assigned to the
-        # volumes/<group-name>/<subvolume-name>/<uuid>/ path when applying
-        # ceph.quota.max_bytes which would assign it with the default value
-        # of is_snapdir_visible flag and right now the child snaprealm
-        # changes are not being compiled and sent to the client by MDS, so until
-        # that gets addressed, as a quick fix apply the vxattr on subvol root
-        # and the uuid path. Once the child snaprealm fix is in place, apply
-        # the vxattr only to subvolume root.
+        # since snapshots are taken at subvolume root, and the parent SnapRealm
+        # changes are not passed on to the child SnapRealm, so, for
+        # subvolumes mounted as NFS shares, just updating the snapshot
+        # visibility vxattr on subvolume root is not enough to invalidate the
+        # NFS-Ganesha cache. This leads to an anomalous behaviour where the old
+        # cache is consulted for checking the snapshot visibility and even if
+        # the visibility is disabled, the snapshots are still visible. The
+        # actual issue is RCA'd and can be tracked at
+        # https://tracker.ceph.com/issues/72400. Until the permanent fix
+        # arrives, resort to unconditionally setting the snapshot visibility
+        # vxattr for subvol v2.
+
         try:
-            subvolume_size = self.fs.getxattr(
-                subvol_v2_path, "ceph.quota.max_bytes").decode('utf-8')
-        except cephfs.NoData:
-            # should be non-sized subvol v2 path
-            pass
-        if int(subvolume_size) > 0:
-            try:
-                self.fs.setxattr(subvol_v2_path, snaps_visibility_vxattr,
-                                 str(value).encode('utf-8'), 0)
-            except cephfs.Error as e:
-                raise VolumeException(-e.args[0], e.args[1])
+            self.fs.setxattr(subvol_v2_path, snaps_visibility_vxattr,
```

Comment:
Also, doesn't snapshot_visibility_get() need to get changed since its fetching from the subvolume parent path.